lib/ansible/modules/windows/win_pagefile.ps1

```diff
@@ -0,0 +1,153 @@
+$maximumSize = Get-AnsibleParam -obj $params -name "maximum_size" -type "int"
+$override =  Get-AnsibleParam -obj $params -name "override" -type "bool" -default $true
+$removeAll = Get-AnsibleParam -obj $params -name "remove_all" -type "bool" -default $false
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "query" -validateset "present","absent","query"
+$systemManaged = Get-AnsibleParam -obj $params -name "system_managed" -type "bool" -default $false
+$testPath = Get-AnsibleParam -obj $params -name "test_path" -type "bool" -default $true
+
+$result = @{
+    changed = $false
+}
+
+if ($removeAll) {
+    $currentPageFiles = Get-WmiObject Win32_PageFileSetting
+    if ($currentPageFiles -ne $null) {
+        $currentPageFiles | Remove-WmiObject -WhatIf:$check_mode | Out-Null
+        $result.changed = $true
+    }
+}
+
+if ($automatic -ne $null) {
+    # change autmoatic managed pagefile 
+    try {
+        $computerSystem = Get-WmiObject -Class win32_computersystem -EnableAllPrivileges
+    
+        if ($computerSystem.AutomaticManagedPagefile -ne $automatic) {
+            $computerSystem.AutomaticManagedPagefile = $automatic
+            if (-not $check_mode) {
+            	$computerSystem.Put() | Out-Null
+            }
+            $result.changed = $true
+        }
+    } catch {
+        Fail-Json $result "Failed to set AutomaticManagedPagefile $($_.Exception.Message)"
+    }
+}
+
+if ($state -eq "absent") {
+    # Remove pagefile
+    if ((Get-Pagefile $fullPath) -ne $null)
+    {
+        try {
+            Remove-Pagefile $fullPath -whatif:$check_mode
+            $result.changed = $true
+        } catch {
+            Fail-Json $result "Failed to remove pagefile $($_.Exception.Message)"
+        }
+    }
+} elseif ($state -eq "present") {   
+    # Remove current pagefile
+    if ($override) {
+        if ((Get-Pagefile $fullPath) -ne $null)
+        {
+            try {
+                Remove-Pagefile $fullPath -whatif:$check_mode
+                $result.changed = $true
+            } catch {
+                Fail-Json $result "Failed to remove current pagefile $($_.Exception.Message)"
+            }
+        }
+    }
+
+    # Make sure drive is accessible
+    if (($test_path) -and (-not (Test-Path "${drive}:"))) {
+        Fail-Json $result "Unable to access '${drive}:' drive"
+    }
+
+    # Set pagefile
+    try {
+        if ((Get-Pagefile $fullPath) -eq $null) {
+            $pagefile = Set-WmiInstance -Class Win32_PageFileSetting -Arguments @{name = $fullPath; InitialSize = 0; MaximumSize = 0} -WhatIf:$check_mode
+            if (-not ($systemManaged -or $check_mode)) {
+                $pagefile.InitialSize = $initialSize
+                $pagefile.MaximumSize = $maximumSize
+                $pagefile.Put() | out-null
+            }
+            $result.changed = $true
+        }
+    } catch {
+        Fail-Json $result "Failed to set pagefile $($_.Exception.Message)"
```

Comment:
This try-catch block is rather large.
I would put 2 different try-catch block around `Set-WmiInstance` and around `$pagefile.Put()`.
Everything else shouldn't fail, and we get more granular error-handling.